shared-libs/rules-engine/src/target-state.js

```diff
@@ -88,52 +92,77 @@ module.exports = {
+          const requestorIds = Object.keys(target.emissions[emissionId]);
+          const filteredInstanceIds = requestorIds.filter(requestorId => !targetEmissionFilter || targetEmissionFilter(target.emissions[emissionId][requestorId]));
           return pick(target.emissions[emissionId], filteredInstanceIds);
         })
-        .filter(emission => emissionClusterFilter(emission))
-        .length;
 
-      minified.value = {
-        pass: countEmissionClusterWith(isPassing),
-        total: countEmissionClusterWith(emission => Object.keys(emission).length > 0),
+        // if there are multiple emissions with the same id emitted by different contacts, disambiguate them
+        .map(emissionsByRequestor => emissionOfLatestRequestor(emissionsByRequestor))
+        .filter(emission => emission);
+        
+      if (!target.passGroupWithCount) {
+        return {
+          pass: relevantEmissions.filter(emission => emission.pass).length,
+          total: relevantEmissions.length,
+        };
+      }
+
+      const countEmissionsByGroup = relevantEmissions
+        .filter(emission => emission.groupBy)
+        .reduce((agg, curr) => {
+          if (!agg[curr.groupBy]) {
+            agg[curr.groupBy] = 0;
+          }
+
+          agg[curr.groupBy]++;
+          return agg;
+        }, {});
+
+      const groups = Object.keys(countEmissionsByGroup);
+      return {
+        pass: groups.filter(group => countEmissionsByGroup[group] >= target.passGroupWithCount).length,
+        total: groups.length,
       };
+    };
+
+    const aggregateTarget = target => {
+      const aggregated = pick(target, ['id', 'type', 'goal', 'translation_key', 'icon', 'subtitle_translation_key']);
+      aggregated.value = scoreTarget(target);
 
-      if (minified.type === 'percent') {
-        minified.value.percent = minified.value.total ? Math.round(minified.value.pass * 100 / minified.value.total) : 0;
+      if (aggregated.type === 'percent') {
+        aggregated.value.percent = aggregated.value.total ? Math.round(aggregated.value.pass * 100 / aggregated.value.total) : 0;
       }
 
-      return minified;
+      return aggregated;
     };
 
-    const isPassing = emission => {
+    const emissionOfLatestRequestor = emissionsByRequestor => {
       let currentOrder = -8640000000000000; // minimum date
-      return Object.keys(emission).reduce((agg, contactId) => {
-        const contactEmission = emission[contactId];
-        if (contactEmission.order > currentOrder) {
-          currentOrder = contactEmission.order;
-          return contactEmission.pass;
+      return Object.keys(emissionsByRequestor).reduce((agg, requestorId) => {
+        const emission = emissionsByRequestor[requestorId];
+        if (emission.order > currentOrder) {
+          currentOrder = emission.order;
+          return emission;
```

Comment:
I think this could be cleaned up by dropping the `currentOrder` and checking for null, eg:

```js
const emission = emissionsByRequestor[requestorId];
if (!agg || emission.order > agg.order) {
  return emission;
}
return agg;
```

You could even make it a ternary for brevity...

```js
const emission = emissionsByRequestor[requestorId];
return (!agg || emission.order > agg.order) ? emission : agg;
```

Then I'd rename some variables for readability...

```js
return Object.keys(emissionsByRequestor).reduce((latest, requestorId) => {
  const current = emissionsByRequestor[requestorId];
  return (!latest || current.order > latest.order) ? current : latest;
}, undefined);
```

Reply:
Went for it. Thanks!